scripts/stubsabot.py

```diff
@@ -170,8 +170,14 @@ async def release_contains_py_typed(release_to_download: PypiReleaseDownload, *,
         raise AssertionError(f"Unknown package type: {packagetype!r}")
 
 
-async def find_first_release_with_py_typed(pypi_info: PypiInfo, *, session: aiohttp.ClientSession) -> PypiReleaseDownload:
+async def find_first_release_with_py_typed(
+    pypi_info: PypiInfo, latest_release: PypiReleaseDownload, *, session: aiohttp.ClientSession
```

Comment:
Technically `latest_release` can be obtained with `pypi_info.get_latest_release()`. But that would result in calling `get_latest_release` twice because it's also used for an early check.

I leave leave this as-is, change the first parameter to expect the iterator of releases instead, or "cache" `get_latest_release` (for example by making it private and exposing it through a property that calls `_get_latest_release` only once)

Reply:
Maybe it could return a bit less early, and use `release_iter = pypi_info.releases_in_descending_order()`. With the `releases_in_descending_order` moving in `find_first_release_with_py_typed`, building an iterator early is probably much better than a redundant call to pypi